### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by picknchew**
Authentication works (Passport and in memory database).
Moved a few things around.
Session and persistent logins still need to be worked on.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -25,18 +28,21 @@ const sessionSettings = {
 // register all partials from directory.
 hbs.registerPartials("./views/partials");
 
-app.set("views", "./views");
+app.set("views", path.join(__dirname, "views"));
```
Comment (gdude2002) [inline app.js:31]: Doesn't this mean the same thing?

File: app.js [hunk 2]
```diff
@@ -25,18 +28,21 @@ const sessionSettings = {
...
-app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.urlencoded({extended: true}));
 // app.use(multer({ dest: './uploads' }));
-app.use(express.static("./public"));
+app.use(express.static(path.join(__dirname, "public")));
+app.use(passport.initialize());
+app.use(passport.session());
 
-Router(app);
+registerRoutes(app, passport);
```
Comment (gdude2002) [inline app.js:45]: You're `require()`-ing a function here, right? But if you're requiring a router, that suggests to me that the router would be a class? Can you add a comment to confirm what exactly you intend?

File: database.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+const User = require("./user.js");
+
+function Database() {
+  var currentId = 1;
+
+  // in memory database.
```
Comment (gdude2002) [inline database.js:6]: I'm assuming you plan to move to some kind of database backend, right?

Comment (picknchew) [reply database.js:6]: Yep, only temporary.

File: app.js (hunk 1 above)
Comment (picknchew) [reply app.js:31]: No they don't. ./ refers to the working directory.

Comment (gdude2002) [reply app.js:31]: That's where this is, isn't it?

File: app.js (hunk 2 above)
Comment (picknchew) [reply app.js:45]: Yes, I'm requiring a function. I changed it from a class so that it seems more appropriate.

File: app.js (hunk 1 above)
Comment (picknchew) [reply app.js:31]: It is, but if you were executing it from somewhere else, it would not be in the current working directory.

Comment (gdude2002) [reply app.js:31]: Right, but you're not being consistent with the usages here. You're `require()`-ing from `./` anyway, so it'll break regardless.

Comment (gdude2002) [reply app.js:31]: It should be a safe assumption that the current working directory is the root of the repo - almost every application requires it.

**Comment by gdude2002** [comment]
You'll probably want to rearrange the templates a little - if possible, have a base template with things that should be present on every page (`<link>`, `<meta>` etc) and inherit from that.

I don't have an example for you for this templating engine, but using Mako with Python, I'd do [something like this](https://github.com/UltrosBot/Ultros-site/blob/falcon/templates/site/base.html), which allows [things like this](https://github.com/UltrosBot/Ultros-site/blob/falcon/templates/site/feeds.html) in other templates.

File: database.js (hunk 3 above)
Comment (gdude2002) [reply database.js:6]: 👍

File: app.js (hunk 2 above)
Comment (gdude2002) [reply app.js:45]: Ah okay, I see what you're doing later on.

This is probably fine for now, but later on you'll probably want a class that will dynamically load up your routes from a configured directory. That'll make sure you don't accidentally miss routes, for example.

**Comment by aramperes** [comment]
@gdude2002 I agree we should improve our template usage, but I think this is out of the scope of this PR. A ticket should be created for this :)

**Comment by gdude2002** [comment]
I agree in part, but this commit is pushing templates in the first place - it's best to start how you mean to finish.

File: app.js (hunk 1 above)
Comment (picknchew) [reply app.js:31]: Actually, require is different. Within require, ./ actually refers to the current directory.

**Review by aramperes — APPROVED** [review]

**Pull request merged**